File: server/apps/assignments/application/use_cases/verify_student_assignment_qr.py
class VerifyStudentAssignmentQrUseCase:

    def __init__(
        self,
        assignment_application_repository,
    ):
        self.assignment_application_repository = (
            assignment_application_repository
        )

    def execute(
        self,
        student_id: str,
        assignment_application_id: str,
        class_section_id: str,
        lesson_id: str,
    ):

        return (
            self.assignment_application_repository
            .verify_student_assignment_qr(
                student_id=student_id,
                assignment_application_id=(
                    assignment_application_id
                ),
                class_section_id=class_section_id,
                lesson_id=lesson_id,
            )
        )

File: client/src/application/assignments/use_cases/verify_student_assignment_qr.js
class VerifyStudentAssignmentQrUseCase {
  constructor(repository) {
    this.repository = repository;
  }

  async execute(req, payload) {
    const studentId = String(payload.studentId || "").trim();

    const assignmentApplicationId = String(
      payload.assignmentApplicationId || "",
    ).trim();

    const classSectionId = String(payload.classSectionId || "").trim();

    const lessonId = String(payload.lessonId || "").trim();

    if (!studentId) {
      throw new Error("Vui lòng nhập mã sinh viên.");
    }

    if (!assignmentApplicationId) {
      throw new Error("Thiếu mã bài tập.");
    }

    if (!classSectionId) {
      throw new Error("Thiếu mã lớp học phần.");
    }

    if (!lessonId) {
      throw new Error("Thiếu mã bài học.");
    }

    return this.repository.verifyStudentAssignmentQr(req, {
      student_id: studentId,
      assignment_application_id: assignmentApplicationId,
      class_section_id: classSectionId,
      lesson_id: lessonId,
    });
  }
}

module.exports = VerifyStudentAssignmentQrUseCase;
